pkg/scalers/azure_pipelines_scaler.go

```diff
@@ -497,9 +498,21 @@ func getCanAgentDemandFulfilJob(jr JobRequest, metadata *azurePipelinesMetadata)
 	if metadata.requireAllDemands {
 		return countDemands == len(demandsInJob) && countDemands == len(demandsInScaler)
 	}
+	if metadata.requireAllDemandsAndIgnoreOthers {
+		return allElementsExist(demandsInScaler, demandsInJob)
```

Comment:
Wondering if we can use a similar logic here as the other returns?
`return countDemands == len(demandsInScaler)`
Does it result in the same?

Reply:
You're right, it probably does. I didn't consider that `countDemands` was the count of all matching demands.